node/graphql/index.ts

```diff
@@ -74,20 +75,47 @@ export const resolvers = {
   Query: {
     getStores: async (_: any, param: any, ctx: any) => {
       const {
-        clients: { hub, sitemap },
+        clients: { hub, sitemap, vbase },
         vtex: { logger },
       } = ctx
 
-      try {
-        sitemap.hasSitemap().then((has: any) => {
-          if (has === false) {
-            sitemap.saveIndex()
-          }
-        })
-      } catch (err) {
-        logger.log(err)
+      const APP_NAME = 'store-locator'
+      const SHCEMA_NAME = 'sitemap'
+
+      const saveInVbase = () => {
+        sitemap
+          .saveIndex()
+          .then((res: any) => {
+            if (res.data.saveIndex) {
+              return vbase.saveJSON(APP_NAME, SHCEMA_NAME, {
+                alreadyHasSitemap: true,
+              })
+            }
+
+            return false
+          })
+          .catch((err) => {
+            logger.log({ error: err, message: 'getStores-saveInBase-error' })
+
+            return false
+          })
       }
```

Comment:
Actually, your function will return always null/undefined because there isn't a returned value.
I suggest you remove the braces in order to return a value, also, your function will return a boolean | Promise;
It'd be better to change the then function to async, so, although the return value, will always return a Promise value;

```suggestion
      const saveInVbase = async () => 
        sitemap
          .saveIndex()
          .then((res: any) => {
            if (res.data.saveIndex) {
              return vbase.saveJSON(APP_NAME, SHCEMA_NAME, {
                alreadyHasSitemap: true,
              })
            }

            return false
          })
          .catch((err) => {
            logger.log({ error: err, message: 'getStores-saveInBase-error' })

            return false
          })
      
```

Reply:
Ok PR updated, thanks.